pkg/controller/nodes/task/k8s/plugin_manager_test.go

```diff
@@ -441,6 +514,69 @@ func TestPluginManager_Abort(t *testing.T) {
 		err = pluginManager.Abort(ctx, tctx)
 		assert.NoError(t, err)
 	})
+
+	t.Run("Abort Plugin has Patch PluginAbortOverride", func(t *testing.T) {
+		tctx := getMockTaskContext(PluginPhaseStarted, PluginPhaseStarted)
+		expectedErr := errors.New("client-side patch error")
+		pluginManager, err := buildPluginWithAbortOverride(
+			ctx,
+			tctx,
+			k8s.AbortBehaviorPatchDefaultResource(k8s.PatchResourceOperation{
+				Patch:   nil,
+				Options: nil,
+			}, false),
+			extendedFakeClient{
+				DeleteError: errors.New(
+					"kubeClient.Delete() should not be called if custom cleanup policy exists"),
+				PatchError: expectedErr,
+			})
+
+		assert.NotNil(t, res)
+		assert.NoError(t, err)
+
+		err = pluginManager.Abort(ctx, tctx)
+		assert.Equal(t, expectedErr, err)
+	})
+
+	t.Run("Abort Plugin has Update PluginAbortOverride", func(t *testing.T) {
+		tctx := getMockTaskContext(PluginPhaseStarted, PluginPhaseStarted)
+		expectedErr := errors.New("client-side update error")
+		pluginManager, err := buildPluginWithAbortOverride(
+			ctx,
+			tctx,
+			k8s.AbortBehaviorUpdateDefaultResource(k8s.UpdateResourceOperation{
+				Options: nil,
+			}, false),
+			extendedFakeClient{
+				DeleteError: errors.New(
+					"kubeClient.Delete() should not be called if custom cleanup policy exists"),
+				UpdateError: expectedErr,
+			})
+
+		assert.NotNil(t, res)
+		assert.NoError(t, err)
+
+		err = pluginManager.Abort(ctx, tctx)
+		assert.Equal(t, expectedErr, err)
+	})
+
+	t.Run("Abort Plugin has Delete PluginAbortOverride", func(t *testing.T) {
```

Comment:
I didn't include tests for AbortBehaviors with a non-nil `Resource` because the existing test framework made it a little difficult (can't make assertions on the calls made to the `extendedFakeClient` as it's not a mock)... I'll brainstorm a bit if there's an easier way to test that.